uppsrc/CtrlCore/CtrlCore.h

```diff
@@ -481,7 +481,9 @@ class Ctrl : public Pte<Ctrl> {
 	};
 
 	Top         *top;
-	int          exitcode;
+#if defined(_M_IX86) // 32bit MSC compiler
+	int			exitcode;
+#endif
```

Comment:
Well - it can not be like this. What is the reason of making this code dedicated to _M_IX86 platform?

Reply:
Klugier:

I am not very familiar with UPP compiler flags. Mirek should be able to come up with the more appropriate one. Theoretically (as far as I know), moving `exitcode` to new location to eliminate padding for 64bit platform (which works for CLANG64 and MSC 64) will not affect size of `Ctrl` on 32 bit platforms. Unfortunately tests indicate for MSBT22(32 bit), size of `Ctrl` will be increased by 8 bytes because of relocation of `exitcode`. This is undesirable. The _M_IX86 flags is used to work around the problem with MSBT22 (and microsoft complier in general). If mirek can use instead the flag the UPP used to differentiate 32bit and 64 bit platform, it will be ideal. `Ctrl` has been tuned perfectly for 32bit platform, we should leave it untouched to avoid introducing unexpected side effects.